File: problems/55. Jump Game/main.py
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n = len(nums)
        max_reach = 0
        for i in range(n):
            if i > max_reach or max_reach >= (n - 1):
                break
            max_reach = max(max_reach, i + nums[i])
        return max_reach >= (n - 1)

File: problems/55. Jump Game/main.go
func canJump(nums []int) bool {
	n := len(nums)
	maxReach := 0
	for i := 0; i < n; i++ {
		if i > maxReach || maxReach >= (n-1) {
			break
		}
		maxReach = max(maxReach, i+nums[i])
	}
	return maxReach >= (n - 1)
}
